packages/nx/src/hasher/task-hasher.ts

```diff
@@ -470,7 +494,7 @@ class TaskHasherImpl {
       const executorPackage = target.executor.split(':')[0];
       const executorNodeName =
         this.findExternalDependencyNodeName(executorPackage);
-      hash = this.hashExternalDependency(executorNodeName);
+      hash = this.hashExternalDependency(targetName, executorNodeName);
```

Comment:
This is not correct. The source is the project name.
```suggestion
      hash = this.hashExternalDependency(projectName, executorNodeName);
```